From cd22ef21e06833945ebf74af3578d6e471a0b464 Mon Sep 17 00:00:00 2001
From: Luis Thomas <luis.thomas@silabs.com>
Date: Tue, 2 Aug 2022 15:26:27 +0200
Subject: [PATCH 1/3] experimental: allow binding with tun interface (based on
 v4_4_3)

ISC DHCP repo can be found here: https://gitlab.isc.org/isc-projects/dhcp

ISC DHCP previously needed to bind to an interface with a MAC address.
In the case of DHCPv6 this is unnecessary. Removing these lines
does not seem to cause problems with DHCPv6.
---
 common/socket.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/common/socket.c b/common/socket.c
index 3953eac3..41bd0a68 100644
--- a/common/socket.c
+++ b/common/socket.c
@@ -577,8 +577,6 @@ if_register6(struct interface_info *info, int do_multicast) {
 	if (req_multi)
 		if_register_multicast(info);
 
-	get_hw_addr(info->name, &info->hw_address);
-
 	if (!quiet_interface_discovery) {
 		if (info->shared_network != NULL) {
 			log_info("Listening on Socket/%d/%s/%s",
@@ -633,8 +631,6 @@ if_register_linklocal6(struct interface_info *info) {
 	info->rfdesc = sock;
 	info->wfdesc = sock;
 
-	get_hw_addr(info->name, &info->hw_address);
-
 	if (!quiet_interface_discovery) {
 		if (info->shared_network != NULL) {
 			log_info("Listening on Socket/%d/%s/%s",
-- 
2.37.2

